### PR TITLE
App to serve livingdocs component

### PR DESCRIPTION
**Pull request by fuenkchen**

### CODE REVIEW

File: client/src/livingdocs-component-app/app.js [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+      this.selectedItem = {
+        id: item.conf._id,
+        toolRuntimeConfig: {
+          displayOptions: {}
+        }
+      };
+    }
+  }
+
+  changeDisplayOptions() {
```
Comment (benib) [inline client/src/livingdocs-component-app/app.js]: I am a little confused by this function and it's name. As far as I get it, it pushes the `selectedItem` to the end of the `selectedItems` array and then triggers a preview loading. But what does this have to do with changing display options? Can you explain?

File: client/src/livingdocs-component-app/app.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+      const result = await this.itemStore.getItems(searchString, numberOfItemsToLoadPerStep, undefined, bookmark);
+      this.itemsLoading = false;
+      return result;
+    } catch (e) {
+      this.messageService.push('error', 'failedLoadingItems');
+    }
+  }
+
+  async loadMore() {
+    const result = await this.loadItems(this.currentSearchString, this.bookmark);
```
Comment (benib) [inline client/src/livingdocs-component-app/app.js]: put this in a try/catch block. otherwise there are errors in nirvana if this fails.

File: client/src/livingdocs-component-app/app.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    const tools = await this.getTools();
```
Comment (benib) [inline client/src/livingdocs-component-app/app.js]: this reloads the tools on every loadMore. I'd load them once in `attached`. or is there a reason why they should get reloaded here that I miss?

File: client/src/livingdocs-component-app/app.js [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+        }
+        this.optionKeys = Object.keys(schema.properties);
+        this.displayOptionsSchema = schema;
+
+        if (!this.selectedItem.toolRuntimeConfig) {
+          this.selectedItem.toolRuntimeConfig = {};
+        }
+
+        if (!this.selectedItem.toolRuntimeConfig.displayOptions) {
+          this.selectedItem.toolRuntimeConfig.displayOptions = {};
```
Comment (benib) [inline client/src/livingdocs-component-app/app.js]: use `generateFromSchema` here. You can import it from `helpers/generateFromSchema.js`. It will generate an object using the default values defined in the passed schema.
If no default values are given for a boolean property it will be `undefined`. But I guess this will work correctly.

File: client/src/livingdocs-component-app/app.js [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+          // we just accept one object containing simple boolean type properties
+          Object.keys(schema.properties).forEach(propertyName => {
+            // maybe delete all non boolean properties here??
+            let displayOption = schema.properties[propertyName];
+            if (displayOption.title) {
+              schema.properties[propertyName].title = this.i18n.tr(`${tool}:${displayOption.title}`);
+            }
+          });
+        }
+        this.optionKeys = Object.keys(schema.properties);
```
Comment (benib) [inline client/src/livingdocs-component-app/app.js]: You do not need this. If you want to walk all object properties in a viewmodel use `for...in`(https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/for...in). If you need to walk them in a view, there is the `keys` value converter so you can do something like this `repeat.for="optionKey of options | keys"`

File: client/src/livingdocs-component-app/app.css [hunk 6]
```diff
@@ -0,0 +1,192 @@
+
+.livingdocs-component-item-container {
+  width: 50%;
+  float: right;
+}
+
+.livingdocs-component-search {
+  width: calc(50% - 1px);
```
Comment (benib) [inline client/src/livingdocs-component-app/app.css]: Set `box-sizing: border-box;` for everything, then you do not have to have the calculation like this as the border is part of the box size then. See https://github.com/nzzdev/Q-editor/blob/dev/client/src/styles/layout.css#L3 and https://github.com/nzzdev/Q-editor/blob/dev/client/src/styles/layout.css#L12-L14

**Comment by benib** [comment]
@fuenkchen can you rebase to current dev? Sorry there where other commits merged first.

**Comment by fuenkchen** [comment]
Done - dev is merged to this branch (did merge instead of rebasing)
